### ISSUE TITLE
Quantizer parameter in mgard_decompress is undocumented

### ISSUE THREAD

**Issue by NAThompson** [CODARcode/MGARD#30]
In the `README.md`, the `quantizer` parameter to `mgard_decompress` is not mentioned, and hence the API is not documented correctly. Also, the meaning of the `quantizer` parameter is unclear.

What does the `quantizer` parameter to `mgard_decompress` do?

In `mgard_test.cpp` line 228, the `quantizer` parameter is labelled as `dummy`, and is not used.

**Comment by qliu21** [comment]
I had a conversation with folks from Brown last week. I was told that any doubles/floats can be passed, and the purpose was to allow C++ overloading to work. 

I believe the APIs need to re-worked in the near future. Since MGARD is used underneath ADIOS at the moment, I wouldn't suggest we change it now, since there was a change in 0.0.0.2, and there was another change after 0.0.0.2.......

**Comment by NAThompson** [comment]
Ok, I'll talk with Scott about this. I'd suggest the following design:

```
namespace mgard {
template<class RandomAccessContainer>
std::vector<char> compress_1d(RandomAccessContainer const & v,
                              typename  RandomAccessContainer::value_type tolerance,
                              typename RandomAccessContainer::value_type s = std::numeric_limits<Real>::infinity());

std::vector<Real> decompress_1d(std::vector<char> && u);

template<class Matrix>
std::vector<char> compress_2d(Matrix const & M,  . . . );
}
```

Assuming you and Scott agree that this would be a positive change,  I'll go ahead and get it done.

**Comment by qliu21** [comment]
@NAThompson Let's discuss this in the next meeting - I am not sure whether we should have compress_1d or _2d, or we should have a generic overloaded signature.

Is it possible to hold this effort off after SC'19? since there will be an ADIOS tutorial in SC19, and they typically will based off master of adios and master of MGARD. If APIs are changed, then we need to re-integrate with ADIOS.

**Comment by NAThompson** [comment]
@qliu21 : We can absolutely hold off. In my mind, this is high priority, but [31](https://github.com/CODARcode/MGARD/issues/31) is way higher.

**Comment by qliu21** [comment]
@NAThompson Thank you and I agree.

**Comment by ben-e-whitney** [comment]
This is fixed in 620b0520e71cc64d96ef3c3be84a2824de398318, included in #89.
